server/socket/index.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-param-reassign */
+const cookie = require('cookie');
+const { WebSocketServer } = require('ws');
+const server = require('..');
+const { verfiyToken } = require('../utils');
+
+const wss = new WebSocketServer({ server });
+
+const assignClient = async (connection, req) => {
+  const cookies = req.headers.cookie;
+  const { token } = cookie.parse(cookies);
+  const decoded = await verfiyToken(token);
+  connection.userId = decoded.id;
+  connection.userName = decoded.username;
+};
+
+const getOnlineUsers = async () => {
+  const onlineUsers = [...wss.clients]
+    .map((client) => ({ userId: client.userId, userName: client.userName }));
+
+  [...wss.clients].forEach((client) => {
+    client.send(JSON.stringify({ onlineUsers }));
+  });
+};
+
+const sendMessage = async (receivedMessage) => {
+  const { text, recipient } = receivedMessage.message;
+  [...wss.clients].forEach((client) => {
+    if (client.userId === recipient) {
+      client.send(JSON.stringify({ text }));
+    }
+  });
+};
+
+wss.on(('connection'), async (connection, req) => {
```

Comment:
take a look for this  link https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener/
* connection is an event ???!
* you must open, close and error events

I noticed the connection occur always and the event need be a string and not require ()